pkg/helmclient/contract.go

```diff
@@ -0,0 +1,82 @@
+package helmclient
+
+import (
+	"time"
+
+	"helm.sh/helm/v3/pkg/release"
+)
+
+// ListOperation defines the contract for a list action
+type ListOperation struct {
+	AllNamespaces bool `json:"all-namespaces,omitempty"`
+	Deployed      bool `json:"deployed,omitempty"`
+	Failed        bool `json:"failed,omitempty"`
+	Pending       bool `json:"pending,omitempty"`
+	Uninstalled   bool `json:"uninstalled,omitempty"`
+	Uninstalling  bool `json:"uninstalling,omitempty"`
+
+	*GlobalFlags
+}
+
+// NewListOperation return an instance of ListOperation
+func NewListOperation() *ListOperation {
+	return &ListOperation{
+		GlobalFlags: &GlobalFlags{},
+	}
+}
+
+// Release represents the Release contract for the interactions with helmclient package
+type Release struct {
+	Name       string         `json:"name"`
+	Namespace  string         `json:"namespace"`
+	Version    int            `json:"version"`
+	Updated    time.Time      `json:"updated_at,omitempty"`
+	Status     release.Status `json:"status"`
+	Chart      string         `json:"chart"`
+	AppVersion string         `json:"app_version"`
+}
+
+// NewRelease returns an instance of Release that wraps release.Release
+func NewRelease(release *release.Release) *Release {
+	return &Release{
+		Name:       release.Name,
+		Namespace:  release.Namespace,
+		Version:    release.Version,
+		Updated:    release.Info.FirstDeployed.Local().Time,
```

Comment:
have methods on struct (or wrapped one) and safeguard with non nil check before accessing to prevent panic